### DISCUSSION TITLE
What is the SDK equivalent of the `aws eks update-kubeconfig` command?

### DISCUSSION THREAD

**Discussion by murshed-panorama** [aws/aws-sdk-ruby#3112] [Q&A] [answered]
Hello! I would like to use the SDK to execute a command like [update-kubeconfig](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/eks/update-kubeconfig.html).

On the command line, it would look something like:

```
aws eks --region us-east-1 update-kubeconfig \
  --name $CLUSTER_NAME \
  --profile $PROFILE_NAME \
  --alias "some_alias" \
  --user-alias "some_user_alias"
```

Does the SDK support this? I reviewed the [EKS Client documentation](https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/EKS/Client.html), but was unable to find a suitable method.

I have the following gems installed

```
gem "aws-sdk-core", "~> 3.208"
gem "aws-sdk-eks", "~> 1.117"
```

**Comment by mullermp** [comment, marked as answer]
https://github.com/aws/aws-cli/blob/develop/awscli%2Fcustomizations%2Feks%2Fupdate_kubeconfig.py

This looks like a customization specifically for CLI and not SDKs. I think you can either keep shelling out to CLI or replicate the logic in Ruby using the eks client.

**Reply by murshed-panorama** [reply]
Ah, that's too bad. I appreciate you providing a link and taking the time to explain. I guess I'll shell out as you recommended. Thank you!
